File: cloudera_training/devsh/exercises/spark-application/countjpgs_project/src/main/scala/solution/CountJPGs.scala
package solution

import org.apache.spark.SparkContext

object CountJPGs {
   def main(args: Array[String]) {
     if (args.length < 1) {
       System.err.println("Usage: solution.CountJPGs <logfile>")
       System.exit(1)
     }

     val sc = new SparkContext()

     val logfile = args(0)
     
     sc.setLogLevel("WARN")
     
     val jpgcount = sc.textFile(logfile).
        filter(line => line.contains(".jpg")).
        count()

     println( "Number of JPG requests: " + jpgcount)

     sc.stop
   }
 }

File: cloudera_training/devsh/exercises/spark-application/python-solution/CountJPGs.py
import sys
from pyspark import SparkContext

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print >> sys.stderr, "Usage: CountJPGs.py <logfile>"
        exit(-1)
        
    sc = SparkContext()
    logfile = sys.argv[1]
    sc.setLogLevel("WARN")
    count = sc.textFile(logfile).filter(lambda line: '.jpg' in line).count()
    print "Number of JPG requests: ", count
    sc.stop()